src/common/gui/base/SingleLineTextField.ts

```diff
@@ -0,0 +1,37 @@
+import m, { Children, ClassComponent, Vnode, VnodeDOM } from "mithril"
+
+export interface SingleLineTextFieldAttrs {
+	disabled?: boolean
+	value: string
+	oninput?: (newValue: string) => unknown
```

Comment:
we should add the comment on when this could be optional (when disabled) otherwise I'll forget :sweat_smile: